app/assets/src/components/views/SampleUploadFlow/SampleUploadFlow.jsx

```diff
@@ -148,20 +157,28 @@ class SampleUploadFlow extends React.Component {
             onDirty={this.metadataChanged}
           />
         )}
-        {this.state.samples && this.state.metadata && (
-          <ReviewStep
-            metadata={this.state.metadata}
-            samples={this.state.samples}
-            uploadType={this.state.uploadType}
-            project={this.state.project}
-            sampleNamesToFiles={this.state.sampleNamesToFiles}
-            hostGenomes={this.props.hostGenomes}
-            visible={this.state.currentStep === "review"}
-            onUploadStatusChange={this.onUploadStatusChange}
-            onStepSelect={this.handleStepSelect}
-            onUploadComplete={this.onUploadComplete}
-          />
-        )}
+        {this.state.samples &&
+          this.state.metadata && (
+            <ReviewStep
+              metadata={this.state.metadata}
+              samples={this.state.samples}
+              uploadType={this.state.uploadType}
+              project={this.state.project}
+              sampleNamesToFiles={this.state.sampleNamesToFiles}
+              hostGenomes={this.state.hostGenomes}
+              visible={this.state.currentStep === "review"}
+              onUploadStatusChange={this.onUploadStatusChange}
+              onStepSelect={this.handleStepSelect}
+              onUploadComplete={this.onUploadComplete}
+            >
+              {this.props.admin && (
+                <HostOrganismMessage
```

Comment:
I'm guessing it's because you needed to pass in `this.props.hostGenomes` (the original host genomes). I would just pass in `originalHostGenomes` as an additional prop. It's still a bit awkward, and some comments would probably be helpful.

Reply:
Passing in as a child node I thought was good (!), on the principle of flatter dependencies and smaller interfaces (props).

In this case the child node could be anything. I can imagine many different messages we may want to go above the continue button.